server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java

```diff
@@ -40,6 +45,160 @@ public void prepareClosedTable() {
+        assertThrows(
+            () -> execute("alter table test open"),
+            isSQLError(
+                is("Relation 'test' unknown"),
+                UNDEFINED_TABLE,
+                BAD_REQUEST,
+                4007
+            )
+        );
+    }
+
+    @Test
+    public void test_open_already_opened_index() {
+        execute("alter table t open");
+        execute("alter table t open");
+    }
+
+    @Test
+    public void test_simple_close_open_with_records() {
+        execute("alter table t open");
+        execute("insert into t values (1), (2)");
+        refresh();
+        execute("select * from t");
+        assertThat(response.rowCount(), is(2L));
+
+        execute("alter table t close");
+        execute("alter table t open");
+        ensureGreen();
+
+        execute("select * from t");
+        assertThat(response.rowCount(), is(2L));
+    }
+
+    @Test
+    public void test_get_translog_stats_after_close_and_open_table() throws Exception {
+        execute("create table test (x int ) with (number_of_replicas = 0) ");
+        long numberOfDocs = randomIntBetween(0, 50);
+        long uncommittedOps = 0;
+        for (long i = 0; i < numberOfDocs; i++) {
+            execute("insert into test values (?)", new Object[]{i});
+            if (rarely()) {
+                execute("optimize table test with (flush = true)");
+                uncommittedOps = 0;
+            } else {
+                uncommittedOps += 1;
+            }
+        }
+        final long uncommittedTranslogOps = uncommittedOps;
+        assertBusy(() -> {
+            execute(
+                "select sum(translog_stats['number_of_operations']), sum(translog_stats['uncommitted_operations']) " +
+                "from sys.shards " +
+                "where table_name = 'test' and primary = true " +
+                "group by table_name, primary");
+            assertThat(response.rowCount(), greaterThan(0L));
+            assertThat(response.rows()[0][0], is(numberOfDocs));
+            assertThat(response.rows()[0][1], is(uncommittedTranslogOps));
+        });
+
+        execute("alter table test close");
+        execute("alter table test open");
+        ensureYellow();
+
+        execute(
+            "select sum(translog_stats['number_of_operations']), sum(translog_stats['uncommitted_operations']) " +
+            "from sys.shards " +
+            "where table_name = 'test' and primary = true " +
+            "group by table_name, primary");
+        assertThat(response.rowCount(), greaterThan(0L));
+        assertThat(response.rows()[0][0], is(numberOfDocs));
+        assertThat(response.rows()[0][1], is(0L));
+    }
+
+    @Test
+    public void test_open_close_is_not_blocked_with_read_and_write_blocks_enabled() {
+        execute("create table test (x int) with (number_of_replicas = 0) ");
+        for (int i = 0; i < between(10, 20); i++) {
+            execute("insert into test values (?)", new Object[]{i});
+        }
```

Comment:
Might be worth changing this to a single bulk insert with `between(10, 20)` to speed up the test case a bit.